### PR TITLE
Add cores for De0 Nano Multicore

### PR DESCRIPTION
**Pull request by stffrdhrn**
Hello, I cleaned up some of these cores from `orpsoc-cores` and here they are.  Sorry not so much comments on the commits I think they are self explanatory. 

The de0_nano-multicore repo is here.  Should we just put the `SoC` cores in fusesoc-cores?  or should they just be standalone projects?

I created the SoC repo there
https://github.com/stffrdhrn/de0_nano-multicore

### CODE REVIEW

**Comment by olofk** [comment]
Thanks a lot. This is a clean patch set, but I still manage to find things to complain about. The thing is that I would like to avoid storing any code in fusesoc-cores. The obvious way to get around this is to create repos for every single one of these small cores (altera_virtual_jtag, gpio, ompic, timer). This works, but I also just figured out that if we don't want to end up with a million repos for all these small cores (there are several more of these tiny core), we could actually just put them in the same repo and fetch them with the url provider if we want. ompic is a bit special as we refer to that in the new arch spec, so I would like to have that in a separate repo under the openrisc org. 

For the rest of the cores,  I have some issues with the versioning.

I don't want to set new versions on the wb_sdram_ctrl and adv_debug_sys unless they have corresponding tags in their repos, as this will be a problem for example when there is an actual release. For adv_debug_sys I just slapped on a new tag (ADS_RELEASE_3_1_0) that we can use instead of the sha. Would be great to see the same for wb_sdram_ctrl.

For mor1kx, it's a bit special since the x.0 versions are considered development releases. We could add `cachable = false` to the provider section to always force a fresh fetch, but that's also a bit annoying since we don't expect it to change that often, so I think mor1kx-5.0 is fine, and we can always do a new mor1kx-5.0-r1 core if we need another one before mor1kx-5.1 is released.

So to sum it up
* I'm picking i2c and mor1kx as is.
* adv_debug_sys is picked but I'm amending the version field in the provider section here.
* wb_sdram_ctrl needs a new tag, or we keep the version at ::wb_sdram_ctrl:0-r2
* ompic code should live in a repo under the openrisc org
* gpio, timer and altera_virtual_jtag should each be put in separate repos, or we can create a new repo for all of them under the fusesoc org. After that we'll add provider sections for them in the core files. I can be proactive here and just do a repo that holds all of them right away

**Comment by olofk** [comment]
ok, I pushed on with this. gpio, altera_virtual_jtag and timer now have a home [here](https://github.com/fusesoc/tiny-cores/) with corresponding core files in fusesoc-cores. All that remains are wb_sdram_ctrl (which is in orpsoc-cores already, so not strictly necessary to add to fusesoc-cores as well)

**Comment by olofk** [comment]
...oh... and ompic

**Comment by stffrdhrn** [comment]
OK I will get started on this. 
 - For `wb_sdram_ctrl` I will update the version and add.  I rather have it here then still depend on orpsoc-cores
 - Ill create an openrisc repo for `ompic`.

Still a question, where should the system cores go?  Do you want the cores in here?

**Comment by olofk** [comment]
Yes, the system cores can be in here, but I want the [.system files gone](https://github.com/olofk/fusesoc/blob/master/doc/migrations.adoc#migrating-from-system-files) first

**Comment by stffrdhrn** [comment]
Closing this, Ill open a new one with the other cores.
